ui/components/MesheryChart.js

```diff
@@ -141,7 +160,8 @@ class MesheryChart extends React.Component {
           tooltip : { show : true, },
         };
         if (!self.props.hideTitle) {
-          self.titleRef.innerText = chartData.options.title.text.join('\n');
+          self.titleRef.innerText = chartData.options.title.text.slice(0,2).join('\n') +"\n"+ chartData.options.title.text[2].split('\n')[0];
```

Comment:
I don't think .innerText is needed here. You can just store in state and use it : 
`self.titleRef = chartData.options.title.text.slice(0,2).join('\n') +"\n"+ chartData.options.title.text[2].split('\n')[0];` 
And don't use callback refs in the JSX part instead just use the data stored in the state. That should work. 
It's not recommended to use ref in place of props or state. They are meant to be used only when it's absolutely necessary.

Reply:
I see that ref is used in the old code too. @meshery/meshery-ui-maintainers is there a particular reason for using refs here, performance-wise or something?

Reply:
I think that was more than 2year old code, I thought not to change logic (**as I am not sure if It will effect some other parts of code**, and discussed with nithish and he said they will start to work on refactoring the code ). There were lot of unwanted logic I see even in chartjs-formatter, may be because it was old code base.

Reply:
Do you recommend me to change to state based approach? I though to use accordion with table to display the data, but It requires refactore the code (as it uses state based approach)